app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -101,9 +142,20 @@ class MessagesController @Inject() (
     implicit request =>
       implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
       (for {
-        upscanResponse              <- parseAndLogUpscanResponse(request.body)
-        downloadUrl                 <- handleUpscanSuccessResponse(upscanResponse)
-        objectSummary               <- objectStoreService.addMessage(downloadUrl, movementId, messageId).asPresentation
+        upscanResponse <- parseAndLogUpscanResponse(request.body)
+        downloadUrl    <- handleUpscanSuccessResponse(upscanResponse)
+        objectSummary <- objectStoreService
+          .addMessage(
+            downloadUrl,
+            movementId,
+            messageId,
+            ObjectStoreFileDirectory(
+              Path.File(s"movements/${movementId.value}/${movementId.value}-${messageId.value}-${fileFormat(clock)}.xml")
+            ),
+            ObjectStoreOwner("common-transit-convention-traders"),
+            ObjectStoreRetentionPeriod(RetentionPeriod.SevenYears)
+          )
```

Comment:
I don't like what you've done here -- you've leaked object-store specific parameters into the controller, when really we should be pushing this sort of thing into the service. I don't think the controller should care about the details of where object-store is putting stuff.

I suspect you did this because you were trying to make a more generic "addMessage" call, but really, all you've done is move all the decisions here, where they shouldn't be. I would much rather there be two functions on the service that set the parameters for us, something like `storeOutgoing` and `storeIncoming`, or something a bit more than that.

Reply:
if we defining storeOutgoing and storeIncoming then we need to make 2 calls to ObjectStoreServiceservice one for to set parameters in storeOutgoing or storeIncoming and then call addMessage. I hope you are fine with that?

Reply:
Why can't whatever `addMessage` does be called by the `store*` methods?